src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/AnnotationUtil.java

```diff
@@ -33,15 +34,28 @@
  */
 public class AnnotationUtil {
     public static List<String> getScopeAnnotations(PsiClass type, Set<String> scopes) {
+        return executeWithExceptionHandling(() ->
+                        // Construct a stream of only the annotations applied to the type that are also
+                        // recognised annotations found in scopes.
+                        Arrays.stream(type.getAnnotations())
+                                .map(annotation -> annotation.getNameReferenceElement().getQualifiedName())
+                                .filter(scopes::contains)
+                                .distinct()
+                                .collect(Collectors.toList()),
+                Collections::emptyList
+        );
+    }
+
+    private static <T> T executeWithExceptionHandling(Supplier<T> action, Supplier<T> fallback) {
         try {
-            // Construct a stream of only the annotations applied to the type that are also
-            // recognised annotations found in scopes.
-            return Arrays.stream(type.getAnnotations()).map(annotation -> annotation.getNameReferenceElement().getQualifiedName())
-                    .filter(scopes::contains).distinct().collect(Collectors.toList());
-        } catch (IndexNotReadyException | ProcessCanceledException | CancellationException e) {
+            return action.get();
+        } catch (ProcessCanceledException e) {
+            //Since 2024.2 ProcessCanceledException extends CancellationException so we can't use multi-catch to keep backward compatibility
+            throw e;
+        } catch (IndexNotReadyException | CancellationException e) {
             throw e;
         } catch (Exception e) {
-            return Collections.<String>emptyList();
+            return fallback.get(); // Return fallback value
         }
     }
```

Comment:
I see this method is private and only called by `getScopeAnnotations()`. Is there some way to combine this with the more widely used utility method you introduced in `ExceptionUtil`? Otherwise this has just introduced more complexity without reducing code duplication in this spot.

Reply:
Yes @mrglavas , I made the changes , please take a look , Thanks